### PR TITLE
ON-39430 # use @oneblink/storage

### PR DESCRIPTION
**Pull request by divporter**

### CODE REVIEW

File: src/lib/one-blink-api.ts [hunk 1]
```diff
@@ -48,6 +51,24 @@ export default class OneBlinkAPI {
...
+    const uploader = new OneBlinkUploader({
+      apiOrigin: OneBlinkAPI.tenant.apiOrigin,
+      region: OneBlinkAPI.tenant.awsRegion,
+      getIdToken: () => {
+        return Promise.resolve(
+          generateJWT(this.accessKey, this.secretKey, this.jwtExpiry),
+        )
+      },
+    })
+    this._oneBlinkUploader
```
Comment (mymattcarroll) [inline src/lib/one-blink-api.ts:68]:
```suggestion
    this._oneBlinkUploader = uploader
```

?

File: src/lib/one-blink-api.ts [hunk 2]
```diff
@@ -48,6 +51,24 @@ export default class OneBlinkAPI {
...
+    }
+    const uploader = new OneBlinkUploader({
+      apiOrigin: OneBlinkAPI.tenant.apiOrigin,
+      region: OneBlinkAPI.tenant.awsRegion,
+      getIdToken: () => {
+        return Promise.resolve(
+          generateJWT(this.accessKey, this.secretKey, this.jwtExpiry),
+        )
+      },
+    })
```
Comment (mymattcarroll) [inline src/lib/one-blink-api.ts:67]: Prob cleaner to just set this in the constructor.

Comment (Zaxist) [reply src/lib/one-blink-api.ts:67]: Agree with Natt

File: package.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
     "@aws-sdk/lib-storage": "^3.521.0",
     "@aws-sdk/s3-request-presigner": "^3.521.0",
     "@oneblink/sdk-core": "^6.1.0-beta.1",
+    "@oneblink/storage": "^1.0.0-beta.6",
```
Comment (mymattcarroll) [inline package.json:15]: Not the latest

File: package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "@aws-sdk/lib-storage": "^3.521.0",
```
Comment (mymattcarroll) [inline package.json:12]: Can this be removed?

File: src/lib/pre-fill-data.ts [hunk 5]
```diff
@@ -1,39 +1,11 @@
-import { S3Client } from '@aws-sdk/client-s3'
-import { Upload } from '@aws-sdk/lib-storage'
-import { PreFillMeta } from '../types'
+import { OneBlinkUploader } from '@oneblink/storage'
```
Comment (mymattcarroll) [inline src/lib/pre-fill-data.ts:1]: Thoughts on deleting this file? Seems just more type checking than needed.

File: src/lib/one-blink-api.ts [hunk 6]
```diff
@@ -46,6 +49,15 @@ export default class OneBlinkAPI {
...
     this.accessKey = accessKey
     this.secretKey = secretKey
+    this.oneBlinkUploader = new OneBlinkUploader({
+      apiOrigin: OneBlinkAPI.tenant.apiOrigin,
+      region: OneBlinkAPI.tenant.awsRegion,
+      getIdToken: () => {
+        return Promise.resolve(
+          `Bearer ${generateJWT(this.accessKey, this.secretKey, this.jwtExpiry)}`,
+        )
+      },
```
Comment (mymattcarroll) [inline src/lib/one-blink-api.ts:59]:
```suggestion
      getIdToken: async () => {
        return `Bearer ${generateJWT(this.accessKey, this.secretKey, this.jwtExpiry)}`
      },
```

its 2023 ;)

Comment (divporter) [reply src/lib/one-blink-api.ts:59]: I'm sticking with 2010

File: package.json (hunk 4 above)
Comment (divporter) [reply package.json:12]: I will checkity check any unused deps

**Pull request merged**